docs/source/reference/iterators.rst

```diff
@@ -17,3 +17,18 @@ Chainer provides some iterators that implement typical strategies to create mini
    chainer.iterators.SerialIterator
    chainer.iterators.MultiprocessIterator
    chainer.iterators.MultithreadIterator
+
+
+Order sampler examples
+======================
+
+An Iterator iterate over a dataset according to ``order``, which is 1-D array of the indices of a dataset.
+Order samplers are functions that are called inside of an iterator to generate an order.
```

Comment:
Just some suggestion regarding the English, how about `An Iterator iterates over a dataset according to an order represented by an 1-D array of indices. Order samplers are callables that are used by those iterators to generate this array.` ?